src/mlpack/core/math/lin_alg.cpp

```diff
@@ -208,3 +208,92 @@ void mlpack::math::RemoveRows(const arma::mat& input,
     }
   }
 }
+
+void mlpack::math::Svec(const arma::mat& input, arma::vec& output)
+{
+  const size_t n = input.n_rows;
+  const size_t n2bar = n * (n + 1) / 2;
+
+  output.zeros(n2bar);
+
+  size_t idx = 0;
+  for (size_t i = 0; i < n; i++)
+  {
+    for (size_t j = i; j < n; j++)
+    {
+      if (i == j)
+        output(idx++) = input(i, j);
+      else
+        output(idx++) = M_SQRT2 * input(i, j);
+    }
+  }
+}
+
+void mlpack::math::Svec(const arma::sp_mat& input, arma::sp_vec& output)
+{
+  const size_t n = input.n_rows;
+  const size_t n2bar = n * (n + 1) / 2;
+
+  output.zeros(n2bar, 1);
+
+  size_t idx = 0;
+
+  for (auto it = input.begin(); it != input.end(); ++it)
```

Comment:
If you use `row_col_iterator` (which, of course, isn't documented on the website... I'll have to make a change and submit it...), which exists for both `mat` and `sp_mat`, it will iterate over all points in `mat` and all nonzero points in `sp_mat`.  Using that, you could condense `Svec` into one templated function, and the speed should be equal to either implementation here.